cmd/podman/common/netflags.go

```diff
@@ -192,42 +204,60 @@ func NetFlagsToNetOptions(opts *entities.NetOptions, flags pflag.FlagSet) (*enti
 		opts.Networks = networks
 	}
 
-	if flags.Changed("ip") || flags.Changed("ip6") || flags.Changed("mac-address") || flags.Changed("network-alias") {
```

Comment:
Please keep the old logic for ip, ip6, mac-address, network-alias. There is no default for the options so this will work correctly. You now removed the important bit for the other options:
```
               // if there is no network we add the default
		if len(opts.Networks) == 0 {
			opts.Networks = map[string]types.PerNetworkOptions{
				"default": {},
			}
		}
```

Reply:
@Luap99 Does this work better?

Reply:
Well that should work too but there is nothing wrong with the current logic so I do not see any reason to change that

Reply:
If someone sets the default on any of the options, then we will pick them up, whether the user changed them or now.

With the new logic, even if the defaults are changed, we will not pass them to the server, unless the user changes them.

Reply:
Right, but you cannot set a default for mac-address, ip, ip6 and network-alias.